src/core/Mixer.cpp

```diff
@@ -368,74 +348,122 @@ void Mixer::deleteChannel( int index )
 	Engine::audioEngine()->doneChangeInModel();
 }
 
+void Mixer::toggleAutoTrackLink(int index)
+{
+	m_mixerChannels[index]->m_autoTrackLinkModel.setValue(! m_mixerChannels[index]->m_autoTrackLinkModel.value());
+}
+
+IntModel * Mixer::getChannelModelByTrack(Track * track)
+{
+	if( track->type() == Track::InstrumentTrack )
+	{
+		InstrumentTrack * inst = (InstrumentTrack *) track;
+		return inst->mixerChannelModel();
+	}
+	else if( track->type() == Track::SampleTrack )
+	{
+		SampleTrack * strk = (SampleTrack *) track;
+		return strk->mixerChannelModel();
+	}
+	return NULL;
+}
+
+bool Mixer::isAutoTrackLinkToggleAllowed(int index)
+{
+	if (mixerChannel( index )->m_autoTrackLinkModel.value()) return true;
 
+	std::vector<int> usedChannelCounts = getUsedChannelCounts();
+	return usedChannelCounts[index] == 1;
+}
 
-void Mixer::moveChannelLeft( int index )
+
+void Mixer::processAssignedTracks(std::function<void(Track * track, IntModel * model, MixerChannel * channel)> process)
 {
-	// can't move master or first channel
-	if( index <= 1 || index >= m_mixerChannels.size() )
+	TrackContainer::TrackList trackList;
+	trackList += Engine::getSong()->tracks();
+	trackList += Engine::patternStore()->tracks();
+
+	for (Track* track: trackList)
+	{
+		IntModel * model = getChannelModelByTrack(track);
+		MixerChannel * channel = NULL;
+		if (model != NULL)
+		{
+			int channelIndex = model->value();
+			if (channelIndex > 0 && channelIndex < m_mixerChannels.size() )
+			{
+				channel = mixerChannel(channelIndex);
+			}
+			process(track,model, channel);
+		}
+	}
+}
+
+std::vector<int> Mixer::getUsedChannelCounts()
+{
+	std::vector<int> used(m_mixerChannels.size(), 0);
+	processAssignedTracks([&used](Track * track, IntModel * model, MixerChannel * channel)
+	mutable {
+		(void) track;
```

Comment:
Is this necessary? I see you've done something like ``(void) <parameter>`` in other places as well, even when the parameter in question isn't used by the lambda.

Reply:
I just wanted to avoid the warning and signal that the parameter is not used by intent but it is not necessary.